tests/python/unittest/test_ndarray.py

```diff
@@ -1099,6 +1099,39 @@ def test_assign_float_value_to_ndarray():
     b[0] = a[0]
     assert same(a, b.asnumpy())
 
+@with_seed()
+def test_ndarray_assignment():
+    H, W = 10, 10
+    a_np = np.random.random((H, W))
+    a_nd = mx.nd.array(a_np)
+    a_nd_id = id(a_nd)
+
+    # assign directly
+    a_np[0] = a_np[1]
+    a_nd[0] = a_nd[1]
+    assert np.allclose(a_np, a_nd.asnumpy())
+    assert id(a_nd) == a_nd_id
```

Comment:
What's the purpose of this check? This seems always true.

Reply:
My mistake. I want to check the memory address of NDArray didn't change.
And `assert np.allclose(a_np, a_nd.asnumpy())` means checking issue #9976.